test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -0,0 +1,228 @@
+		zoneUpdated, err := findZoneByName(nrtUpdated, zoneInitial.Name)
+		if err != nil {
+			klog.Errorf("missing updated zone %q: %v", zoneInitial.Name, err)
+			return false, err
+		}
+		ok, what, err := checkEqualResourcesInfo(nrtInitial.Name, zoneInitial.Name, zoneInitial.Resources, zoneUpdated.Resources)
+		if err != nil {
+			klog.Errorf("error checking zone %q: %v", zoneInitial.Name, err)
+			return false, err
+		}
+		if !ok {
+			klog.Infof("node %q zone %q resource %q is different", nrtInitial.Name, zoneInitial.Name, what)
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func CheckZoneConsumedResourcesAtLeast(nrtInitial, nrtUpdated nrtv1alpha1.NodeResourceTopology, required corev1.ResourceList) (string, error) {
+	for idx := 0; idx < len(nrtInitial.Zones); idx++ {
+		zoneInitial := &nrtInitial.Zones[idx] // shortcut
+		zoneUpdated, err := findZoneByName(nrtUpdated, zoneInitial.Name)
+		if err != nil {
+			klog.Errorf("missing updated zone %q: %v", zoneInitial.Name, err)
+			return "", err
+		}
+		ok, err := checkConsumedResourcesAtLeast(zoneInitial.Resources, zoneUpdated.Resources, required)
+		if err != nil {
+			klog.Errorf("error checking zone %q: %v", zoneInitial.Name, err)
+			return "", err
+		}
+		if ok {
+			klog.Infof("match for zone %q", zoneInitial.Name)
+			return zoneInitial.Name, nil
+		}
+	}
+	return "", nil
+}
+
+func checkEqualResourcesInfo(nodeName, zoneName string, resourcesInitial, resourcesUpdated []nrtv1alpha1.ResourceInfo) (bool, string, error) {
+	for _, res := range resourcesInitial {
+		initialQty := res.Available
+		updatedQty, ok := findResourceAvailableByName(resourcesUpdated, res.Name)
+		if !ok {
+			return false, res.Name, fmt.Errorf("resource %q not found in the updated set", res.Name)
+		}
+		if initialQty.Cmp(updatedQty) != 0 {
+			klog.Infof("node %q zone %q resource %q initial=%v updated=%v", nodeName, zoneName, res.Name, initialQty, updatedQty)
+			return false, res.Name, nil
+		}
+	}
+	return true, "", nil
+}
+
+func checkConsumedResourcesAtLeast(resourcesInitial, resourcesUpdated []nrtv1alpha1.ResourceInfo, required corev1.ResourceList) (bool, error) {
+	for resName, resQty := range required {
+		initialQty, ok := findResourceAvailableByName(resourcesInitial, string(resName))
+		if !ok {
+			return false, fmt.Errorf("resource %q not found in the initial set", string(resName))
+		}
+		updatedQty, ok := findResourceAvailableByName(resourcesUpdated, string(resName))
+		if !ok {
+			return false, fmt.Errorf("resource %q not found in the updated set", string(resName))
+		}
+		expectedQty := initialQty.DeepCopy()
+		expectedQty.Sub(resQty)
+		ret := updatedQty.Cmp(expectedQty)
+		if ret > 0 {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func ResourcesFromPod(pod corev1.Pod) corev1.ResourceList {
+	res := make(corev1.ResourceList)
+	for idx := 0; idx < len(pod.Spec.Containers); idx++ {
+		cnt := &pod.Spec.Containers[idx] // shortcut
+		for resName, resQty := range cnt.Resources.Limits {
```

Comment:
Do we care only about limits?
What if the pod is burstable and only have requests

Reply:
for the current use cases yes :) I'll rename the function to make evident we deal only with the Guaranteed pods atm.